src/units/rtr.rs

```diff
@@ -290,20 +290,18 @@ where
         let mut target = Target::new(component.name().clone());
         component.register_metrics(metrics.clone());
         let mut this = Self::new(connect, retry, metrics);
-        gate.update_status(UnitStatus::Stalled).await;
         loop {
             debug!("Unit {}: Connecting ...", target.name);
             let mut client = match this.connect(target, &mut gate).await {
```

Comment:
I think this could be simplified with a `let Ok(mut client) = this.connect(target, &mut gate).await else { ... }` style.

Reply:
Certainly, but I find that very confusing and would rather stick to the match in this case.